homeassistant/components/tuya/climate.py

```diff
@@ -231,6 +231,11 @@ def __init__(
                 description.switch_only_hvac_mode,
             ]
 
+        # determine HVAC mode DPCode to use POWER_SWITCH and not SWITCH if available
+        self._hvac_mode_dp_name = DPCode.SWITCH
+        if DPCode.POWER_SWITCH in self.device.function:
```

Comment:
A second opinion is needed here. Are these two opinions exclusive? What should we do if a device supports both?

@jmcollin78 Does your device include `DPCode.SWITCH` in `self.device.function`?

Maybe we could extend the if statement to 
```suggestion
        if DPCode.POWER_SWITCH in self.device.function and not DPCode.SWITCH in self.device.function:
```

Before applying this suggestion, wait for another opinion.

Reply:
No. My device have only DPCode.POWER in function.

Reply:
The device can be configured in standard instruction set (the previous "mode") or original DP instruction set which include the "POWER". The device cannot have both.
See more explaination here if needed : https://github.com/home-assistant/core/issues/78815

This PR fixes this issue.

Reply:
Thus, your last proposal will not change anything.

Reply:
Device are only in one mode possible : POWER or SWITCH. We can change this putting the device into Native CP mode on the TuYA cloud. This is explain here: https://github.com/home-assistant/core/issues/78815

So for me, your last proposal is useless.